### PR TITLE
Fix track color library inline editing

### PR DESCRIPTION
**Pull request by Holzhaus**
Fixes the issue reported here: https://github.com/mixxxdj/mixxx/pull/2470#issuecomment-590589277

### CODE REVIEW

File: src/library/basesqltablemodel.cpp [hunk 1]
```diff
@@ -910,6 +912,8 @@ Qt::ItemFlags BaseSqlTableModel::readWriteFlags(
             index.row(), fieldIndex(ColumnCache::COLUMN_LIBRARYTABLE_BPM_LOCK))
                 .data().toBool();
         return locked ? defaultFlags : defaultFlags | Qt::ItemIsEditable;
+    } else if (column == fieldIndex(ColumnCache::COLUMN_LIBRARYTABLE_COLOR)) {
+        return defaultFlags;
```
Comment (uklotzde) [inline src/library/basesqltablemodel.cpp]: Please add this case to the other columns that all return `defaultFlags`

File: src/library/basesqltablemodel.cpp [hunk 2]
```diff
@@ -723,6 +723,10 @@ QVariant BaseSqlTableModel::data(const QModelIndex& index, int role) const {
     // role
     switch (role) {
         case Qt::ToolTipRole:
+            if (column == fieldIndex(ColumnCache::COLUMN_LIBRARYTABLE_COLOR)) {
```
Comment (uklotzde) [inline src/library/basesqltablemodel.cpp]: Wouldn't QVariant() work be more appropriate here?

Comment (uklotzde) [reply src/library/basesqltablemodel.cpp]: This works for me:

```
const auto color = mixxx::RgbColor::optional(value);
if (color) {
    value = mixxx::toQColor(color).name();
} else {
    value = QString();
}
```

Comment (uklotzde) [reply src/library/basesqltablemodel.cpp]: Maybe add a reusable, free function to `rgbcolor.h`:
`QString nullableName(mixxx::RgbColor::optional_t color);`

Please note that QColor().name() seems to return "#000000", i.e. black although undefined/invalid. So we should test this new function!

Not sure about the naming, but at least it is concise and easy to remember.

**Comment by uklotzde** [comment]
@Holzhaus GitHub doesn't offer me the option to open a PR for you repo:

https://github.com/uklotzde/mixxx/tree/trackcolor-fix

File: src/library/basesqltablemodel.cpp (hunk 2 above)
Comment (Holzhaus) [reply src/library/basesqltablemodel.cpp]: Do you want me to do this as part of this PR?

Comment (uklotzde) [reply src/library/basesqltablemodel.cpp]: Hold on. Let's first merge this PR and I will publish a separate PR then.

**Review by uklotzde — APPROVED** [review]

File: src/library/basesqltablemodel.cpp [hunk 3]
```diff
@@ -723,6 +723,15 @@ QVariant BaseSqlTableModel::data(const QModelIndex& index, int role) const {
...
+                const auto color = mixxx::RgbColor::optional(value);
+                if (color) {
+                    value = mixxx::toQColor(color).name();
+                } else {
+                    value = QString();
+                }
+            }
+            [[fallthrough]];
```
Comment (daschuer) [inline src/library/basesqltablemodel.cpp]: This causes unfortunately a warning. 
Please use M_FALLTHROUGH_INTENDED

Comment (uklotzde) [reply src/library/basesqltablemodel.cpp]: Fixed in #2518

Comment (daschuer) [reply src/library/basesqltablemodel.cpp]: Thank you.

**Pull request merged**